software/base/src/main/java/org/apache/brooklyn/entity/software/base/lifecycle/MachineLifecycleEffectorTasks.java

```diff
@@ -124,7 +124,7 @@
     public static final ConfigKey<Duration> STOP_PROCESS_TIMEOUT = ConfigKeys.newConfigKey(Duration.class,
             "process.stop.timeout", "How long to wait for the processes to be stopped; use null to mean forever", Duration.TWO_MINUTES);
 
-    protected final MachineInitTasks machineInitTasks = new MachineInitTasks();
+    protected final transient MachineInitTasks machineInitTasks = new MachineInitTasks();
```

Comment:
Thinking more about this it will do more damage than good:
  * `MachineInitTasks` doesn't have any properties (hence state), so is lightweight to serialize
  * On rebind the constructor of objects in config bag is not called by xstream, so this will be null. In most cases where the default value of the config key `SoftwareProcess.LIFECYCLE_EFFECTOR_TASKS` is used it will work fine, but when the user sets a value and it is serialized things will fall apart.

Tried adding `DoNothingSoftwareProcess` to the `BrooklynMementoPersisterInMemorySizeIntegrationTest`, but the OOME didn't surface. Will continue looking.